keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ResourceRepository.kt

```diff
@@ -129,6 +147,12 @@ interface ResourceRepository : PeriodicallyCheckedRepository<Resource<out Resour
    */
   fun appendHistory(event: ResourceEvent)
 
+  /**
+   * Records an event associated with an application.
+   * TODO: adding this here as there's no ApplicationRepository or EventRepository, but might want to move it.
+   */
+  fun appendHistory(event: ApplicationEvent)
```

Comment:
Is there an advantage to having separate `appendHistory` calls for the different subclasses of `PersistentEvent` versus just having one method for the base class:

```kt
fun appendHistory(event: PersistentEvent)
```

Reply:
I actually tried to merge them, but the issue I ran into is that we are dynamically determining the value to insert in the `uid` column for `ResourceEvents` by doing a select into the `resources` table, which prevents adding events for resources that don't exist in the database (which is a good thing since we don't have FKs anywhere). We could add the `uid` to insert as a parameter, but I think this would complicate life unnecessarily for callers.